app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt

```diff
@@ -286,31 +289,78 @@ open class FullScreenPlayer : AbstractPlayerFragment() {
             player.getCurrentPreferredSubtitle() == null
     }
 
+    private fun restoreOrientationWithSensor(activity: Activity){
+        val currentOrientation = activity.resources.configuration.orientation
+        var orientation = 0
+        when (currentOrientation) {
+            Configuration.ORIENTATION_LANDSCAPE ->
+                orientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+
+            Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED ->
+                dynamicOrientation(activity)
+
+            Configuration.ORIENTATION_PORTRAIT ->
+                orientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT
+        }
+        activity.requestedOrientation = orientation
+    }
+
+    private fun toggleOrientationWithSensor(activity: Activity){
+        val currentOrientation = activity.resources.configuration.orientation
+        var orientation = 0
+        when (currentOrientation) {
+            Configuration.ORIENTATION_LANDSCAPE ->
+                orientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT
+
+            Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED ->
+                dynamicOrientation(activity)
+
+            Configuration.ORIENTATION_PORTRAIT ->
+                orientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+        }
+        activity.requestedOrientation = orientation
+    }
+
     open fun lockOrientation(activity: Activity) {
         val display =
             (activity.getSystemService(Context.WINDOW_SERVICE) as WindowManager).defaultDisplay
         val rotation = display.rotation
         val currentOrientation = activity.resources.configuration.orientation
         var orientation = 0
         when (currentOrientation) {
-            Configuration.ORIENTATION_LANDSCAPE -> orientation =
-                if (rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_90) ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE else ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE
-
-            Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED, Configuration.ORIENTATION_PORTRAIT -> ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
-            //Configuration.ORIENTATION_PORTRAIT -> orientation =
-            //    if (rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_270) ActivityInfo.SCREEN_ORIENTATION_PORTRAIT else ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT
+            Configuration.ORIENTATION_LANDSCAPE ->
+                orientation =
+                    if (rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_90)
+                        ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+                    else
+                        ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE
+
+            Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED ->
+                dynamicOrientation(activity)
```

Comment:
This will cause a bug as orientation will still be zero and 0 = SCREEN_ORIENTATION_LANDSCAPE, so this function will be called then it will call `activity.requestedOrientation = SCREEN_ORIENTATION_LANDSCAPE` right under this

Reply:
you have the same problem in the other functons

Reply:
This is how it was -> `Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED, Configuration.ORIENTATION_PORTRAIT -> ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE`
Is this doing anything? there is no assignment.

Now in function i have added -> `activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE`

So what should be the change? should I keep it how it was? how to test this Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED?

Reply:
What I mean is that if you read the code 
![image](https://github.com/recloudstream/cloudstream/assets/147925818/2f85f82e-ae89-407d-a569-8e2d0042f031)
Then it will set `activity.requestedOrientation = orientation` and `orientation = 0` when `currentOrientation = Configuration.ORIENTATION_SQUARE, Configuration.ORIENTATION_UNDEFINED` so dynamicOrientation wont do anything as it will be overridden 2 lines later

Reply:
In this case I think dynamicOrientation should return what orientation it should be and then set `orientation = dynamicOrientation(activity)` inside the when statments

Reply:
ohk I missed that

Reply:
@fire-light42 thanks. Pls chk now.